### ISSUE TITLE
Gradle plugin fails on Windows/Linux with Gradle 6.6

### ISSUE THREAD

**Issue by abrooksv** [JetBrains/rd#139]
It was working on Gradle 6.5

Error:

```
> Task :jetbrains-rider:generateAwsProjectModel
Reading hash from D:\aws-toolkit-jetbrains\jetbrains-rider\build\rdgen\.rdgen
Hashes are different at key 'file: C:\Users\<user>\.gradle\caches\modules-2\files-2.1\com.jetbrains.rd\rd-gen\0.193.146\38e03009032d0ea8ff19183cb90afdca3e6f09e3\rd-gen-0.193.146.jar', oldHash: 2020-05-19_09:12:06.2
80, newHash:
Temporary folder created: C:\Users\<user>\AppData\Local\Temp\rdgen-16595563179377353196
Searching for Kotlin compiler
Compiling sources from 'D:\aws-toolkit-jetbrains\jetbrains-rider\protocol\model\project' to 'C:\Users\<user>\AppData\Local\Temp\rdgen-16595563179377353196'
User classpath: 'D:\aws-toolkit-jetbrains\jetbrains-rider\build\riderRD-2019.3.4\lib\rd\rider-model.jar'
Rdgen default classpath: 'C:\Users\<user>\.gradle\caches\jars-8\c0a48cb2f5b7c3e7dd64ce031908edd4\rd-gen-0.193.146.jar;C:\Users\<user>\.gradle\caches\jars-8\b286736391573ccfb4d4a500279a066b\kotlin-stdlib-jdk8-1.3.
70.jar;C:\Users\<user>\.gradle\caches\jars-8\1e6c5a27e924f3498df588a40253ce5a\kotlin-stdlib-jdk7-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\1be4573a90e903c998485b49504b2e84\kotlin-reflect-1.3.50.jar;C:\User
s\<user>\.gradle\caches\jars-8\a87785aaeadbb9423aaef6ceba52a690\kotlin-stdlib-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\6318226462ad1e7ffc43ccf52d56b099\kotlin-stdlib-common-1.3.70.jar;C:\Users\<user>\.g
radle\caches\jars-8\a4fe510e7beb8b97871b47d87c0bd50b\kotlin-compiler-embeddable-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\dde97575a599bd3e140dd938a7aed4fa\kotlin-reflect-1.3.72.jar;C:\Users\<user>\.gradle\
caches\jars-8\6e1c96bc5e566162587181c85dab9756\kotlin-scripting-jvm-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\b7970a59300ebc4e2f7887783c16e693\kotlin-scripting-common-1.3.70.jar;C:\Users\<user>\.gradle\cac
hes\jars-8\768cf1021ee3bdc425b9b2ae1289f2f1\kotlin-stdlib-jdk8-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\421b0c68a25f5d684ea66be525dfcf66\kotlin-stdlib-jdk7-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8
\70d15c94187de212f9a927dc262ddd63\kotlin-stdlib-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\18e6bc6874baa1a066bce674522f54bf\kotlin-stdlib-common-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\b2c292c88723
c40ac5c567f0ca5e67c6\kotlin-script-runtime-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\65abeebf44db8fc52a0efcfb702739ec\kotlin-compiler-1.3.50.jar;C:\Users\<user>\.gradle\caches\jars-8\e6f0c38292333779074fe9
18cc0ffc34\kotlin-script-runtime-1.3.50.jar'
Resulting kotlinc classpath: 'D:\aws-toolkit-jetbrains\jetbrains-rider\build\riderRD-2019.3.4\lib\rd\rider-model.jar;C:\Users\<user>\.gradle\caches\jars-8\c0a48cb2f5b7c3e7dd64ce031908edd4\rd-gen-0.193.146.jar;C:\Us
ers\<user>\.gradle\caches\jars-8\b286736391573ccfb4d4a500279a066b\kotlin-stdlib-jdk8-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\1e6c5a27e924f3498df588a40253ce5a\kotlin-stdlib-jdk7-1.3.70.jar;C:\Users\alfred
br\.gradle\caches\jars-8\1be4573a90e903c998485b49504b2e84\kotlin-reflect-1.3.50.jar;C:\Users\<user>\.gradle\caches\jars-8\a87785aaeadbb9423aaef6ceba52a690\kotlin-stdlib-1.3.70.jar;C:\Users\<user>\.gradle\caches\j
ars-8\6318226462ad1e7ffc43ccf52d56b099\kotlin-stdlib-common-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\a4fe510e7beb8b97871b47d87c0bd50b\kotlin-compiler-embeddable-1.3.70.jar;C:\Users\<user>\.gradle\caches\j
ars-8\dde97575a599bd3e140dd938a7aed4fa\kotlin-reflect-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\6e1c96bc5e566162587181c85dab9756\kotlin-scripting-jvm-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\b7970a
59300ebc4e2f7887783c16e693\kotlin-scripting-common-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\768cf1021ee3bdc425b9b2ae1289f2f1\kotlin-stdlib-jdk8-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\421b0c68a25
f5d684ea66be525dfcf66\kotlin-stdlib-jdk7-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\70d15c94187de212f9a927dc262ddd63\kotlin-stdlib-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\18e6bc6874baa1a066bce67452
2f54bf\kotlin-stdlib-common-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\b2c292c88723c40ac5c567f0ca5e67c6\kotlin-script-runtime-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\65abeebf44db8fc52a0efcfb702739e
c\kotlin-compiler-1.3.50.jar;C:\Users\<user>\.gradle\caches\jars-8\e6f0c38292333779074fe918cc0ffc34\kotlin-script-runtime-1.3.50.jar'
kotlinc -cp D:\aws-toolkit-jetbrains\jetbrains-rider\build\riderRD-2019.3.4\lib\rd\rider-model.jar;C:\Users\<user>\.gradle\caches\jars-8\c0a48cb2f5b7c3e7dd64ce031908edd4\rd-gen-0.193.146.jar;C:\Users\<user>\.grad
le\caches\jars-8\b286736391573ccfb4d4a500279a066b\kotlin-stdlib-jdk8-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\1e6c5a27e924f3498df588a40253ce5a\kotlin-stdlib-jdk7-1.3.70.jar;C:\Users\<user>\.gradle\caches\
jars-8\1be4573a90e903c998485b49504b2e84\kotlin-reflect-1.3.50.jar;C:\Users\<user>\.gradle\caches\jars-8\a87785aaeadbb9423aaef6ceba52a690\kotlin-stdlib-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\6318226462ad
1e7ffc43ccf52d56b099\kotlin-stdlib-common-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\a4fe510e7beb8b97871b47d87c0bd50b\kotlin-compiler-embeddable-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\dde97575a599
bd3e140dd938a7aed4fa\kotlin-reflect-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\6e1c96bc5e566162587181c85dab9756\kotlin-scripting-jvm-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\b7970a59300ebc4e2f788778
3c16e693\kotlin-scripting-common-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\768cf1021ee3bdc425b9b2ae1289f2f1\kotlin-stdlib-jdk8-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\421b0c68a25f5d684ea66be525dfc
f66\kotlin-stdlib-jdk7-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\70d15c94187de212f9a927dc262ddd63\kotlin-stdlib-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\18e6bc6874baa1a066bce674522f54bf\kotlin-stdl
ib-common-1.3.72.jar;C:\Users\<user>\.gradle\caches\jars-8\b2c292c88723c40ac5c567f0ca5e67c6\kotlin-script-runtime-1.3.70.jar;C:\Users\<user>\.gradle\caches\jars-8\65abeebf44db8fc52a0efcfb702739ec\kotlin-compiler-
1.3.50.jar;C:\Users\<user>\.gradle\caches\jars-8\e6f0c38292333779074fe918cc0ffc34\kotlin-script-runtime-1.3.50.jar -d C:\Users\<user>\AppData\Local\Temp\rdgen-16595563179377353196 -jvm-target 1.8 D:\aws-toolkit-j
etbrains\jetbrains-rider\protocol\model\project
warning: unable to find kotlin-stdlib.jar in the Kotlin home directory. Pass either '-no-stdlib' to prevent adding it to the classpath, or the correct '-kotlin-home'
warning: unable to find kotlin-script-runtime.jar in the Kotlin home directory. Pass either '-no-stdlib' to prevent adding it to the classpath, or the correct '-kotlin-home'
warning: unable to find kotlin-reflect.jar in the Kotlin home directory. Pass either '-no-reflect' or '-no-stdlib' to prevent adding it to the classpath, or the correct '-kotlin-home'
exception: java.lang.NoClassDefFoundError: javaslang/?
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:550)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:550)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:550)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at org.jetbrains.kotlin.resolve.calls.smartcasts.DataFlowInfoImpl.<clinit>(DataFlowInfoImpl.kt:341)
        at org.jetbrains.kotlin.resolve.calls.smartcasts.DataFlowInfoFactory.<clinit>(DataFlowInfo.kt:104)
        at org.jetbrains.kotlin.resolve.calls.smartcasts.DataFlowInfo$Companion.<clinit>(DataFlowInfo.kt:98)
        at org.jetbrains.kotlin.resolve.calls.smartcasts.DataFlowInfo.<clinit>(DataFlowInfo.kt)
        at org.jetbrains.kotlin.resolve.LazyTopDownAnalyzer.analyzeDeclarations$default(LazyTopDownAnalyzer.kt:60)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:112)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:82)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:554)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:81)
        at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:107)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:545)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:176)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:163)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:51)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:85)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:43)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:104)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:82)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:50)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.jetbrains.rd.generator.nova.RdGen.compile0(RdGen.kt:106)
        at com.jetbrains.rd.generator.nova.RdGen.compileDsl(RdGen.kt:146)
        at com.jetbrains.rd.generator.nova.RdGen.run(RdGen.kt:284)
        at com.jetbrains.rd.generator.gradle.RdgenTask.run(RdgenPlugin.kt:116)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:570)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:395)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:387)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:84)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:555)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:538)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:279)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:268)
        at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$1(ExecuteStep.java:33)
        at java.base/java.util.Optional.orElseGet(Optional.java:369)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:33)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
        at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
        at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
        at java.base/java.util.Optional.map(Optional.java:265)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
        at java.base/java.util.Optional.orElseGet(Optional.java:369)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:195)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:187)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:409)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:399)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:157)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:242)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:150)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:94)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: javaslang.?
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 157 more

Compilation finished in 3515 ms
Compilation failed. Return code: 2(INTERNAL_ERROR).

Usage: java -cp [CLASSPATH] com.jetbrains.rd.generator.nova.MainKt [OPTION]...
RD Generator, v1.07. Search for inheritors of 'com.jetbrains.rd.generator.nova.Toplevel' and generate sources according generators: inheritors of 'com.jetbrains.rd.generator.nova.IGenerator'.

Options:
  -s, --source=<source>     Folders with dsl .kt files. If not present, scan classpath for inheritors of 'com.jetbrains.rd.generator.nova.Toplevel'
  -h, --hash-folder=<hash-folder>Folder to store hash file '.rdgen' for incremental generation; default: C:\Users\<user>\.gradle\daemon\6.6
      --compiled=<compiled> Folder for compiled dsl. Temporary folder is created if option is not specified.
  -c, --compiler-classpath=<compiler-classpath>Classpath for kotlin compiler. You must specify it if you referenced something from your dsl
  -f, --force               Suppress incremental generation.
  -x, --clear               Clear output folder before generation (if it is not incremental)
  -p, --packages=<packages> Java package names to search toplevels, delimited by ','. Example: com.jetbrains.rd.model.nova; default: com,org
      --filter=<filter>     Filter generators by searching regular expression inside generator class simple name (case insensitive). Example: kotlin|csharp|cpp
  -v, --verbose             Verbose output
```

I tried switching to `kotlin-compler-embedded` as a dependency but that did not help.

I also tried a custom Gradle task that uses `project.javaexec` and custom `configuration` to model the Rdgen dependency to invoke Rdgen as a CLI in order to isolate the classpath from Gradle, but it looks like the `generators` block was never modeled as a CLI argument.

**Comment by ForNeVeR** [comment]
We're struggling with the same problem ourselves; right now working on a fix.

**Comment by abrooksv** [comment]
If it is any help here is the custom RdPlugn I was trying to build that would have forked RdGen into its own process

```
import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.file.Directory
import org.gradle.api.model.ObjectFactory
import org.gradle.api.provider.ListProperty
import org.gradle.api.provider.Property
import org.gradle.api.tasks.Input
import org.gradle.api.tasks.InputFiles
import org.gradle.api.tasks.SourceTask
import org.gradle.api.tasks.TaskAction
import org.gradle.kotlin.dsl.getByName
import org.gradle.kotlin.dsl.listProperty
import org.gradle.kotlin.dsl.maven
import org.gradle.kotlin.dsl.property
import org.gradle.workers.WorkParameters
import org.gradle.workers.WorkerExecutor
import javax.inject.Inject

class RdGenPlugin : Plugin<Project> {
    companion object {
        const val RDGEN_CONFIGURATION_NAME = "rdgen"
        const val PLUGIN_EXTENSION_NAME = "rdgen"
    }

    private lateinit var project: Project

    override fun apply(target: Project) {
        this.project = target

        addConfiguration()
        addExtension()

        configureRdGenDependency()
    }

    private fun addConfiguration() {
        val rdGenConf = project.configurations.create(RDGEN_CONFIGURATION_NAME)
        rdGenConf.isVisible = false
        rdGenConf.isTransitive = true
        rdGenConf.description = "The classpath of JetBrains RD"

        project.repositories.maven("https://www.myget.org/F/rd-snapshots/maven/")
    }

    private fun addExtension() {
        project.extensions.create(PLUGIN_EXTENSION_NAME, RdGenPluginExtension::class.java)
    }

    private fun configureRdGenDependency() {
        val extension = project.extensions.getByName<RdGenPluginExtension>(PLUGIN_EXTENSION_NAME)
        project.configurations.getByName(RDGEN_CONFIGURATION_NAME) {
            defaultDependencies {
                add(project.dependencies.create("com.jetbrains.rd:rd-gen:${extension.version}"))
            }
        }
    }
}

class RdGenPluginExtension(objects: ObjectFactory) {
    val version = objects.property<String>()
}

class RdGenTask @Inject constructor(private val workerExecutor: WorkerExecutor) : SourceTask() {
    @InputFiles
    val classpath = project.objects.fileCollection()

    @Input
    val generators = project.objects.listProperty<Generator>()

    @TaskAction
    fun run() {
        project.javaexec {
            this.classpath = this@RdGenTask.classpath
            this.mainClass.set("com.jetbrains.rd.generator.nova.RdGen")
        }
    }
}

interface RdGenParameters : WorkParameters {
    val generators: ListProperty<Generator>
}

interface Generator {
    @Input
    fun getLanguage(): Property<String>

    @Input
    fun transform(): Property<String>

    @Input
    fun root(): Property<Directory>

    @Input
    fun namespace(): Property<String>

    @Input
    fun directory(): Property<Directory>
}
```

Thought was to split RdGen from the RdPlugin so that it could be resolved as it own dependency and isolated from the rest of Gradle by invoking it as a forked Java process, but I hit a roadblock when I found that gradleGenerationSpecs was not added as a CLI option:
https://github.com/JetBrains/rd/blob/master/rd-kt/rd-gen/src/main/kotlin/com/jetbrains/rd/generator/nova/RdGen.kt#L61

None of the above was tested since I gave up when I fond it impossible to continue the port

**Comment by ForNeVeR** [comment]
Great minds think alike; it is the same we're currently trying to do. It seems to work, but there're some other problems. Currently, there's a hope we'll fix the remaining issues tomorrow (it's already the third time in a row they're supposed to be fixed "tomorrow" though).

**Comment by ForNeVeR** [comment]
@abrooksv, please try to migrate to 0.203.144, and let us know if you have any new problems.

**Comment by abrooksv** [comment]
@ForNeVeR Thank you, it seems to work.

**Comment by mfilippov** [comment]
@abrooksv we released new version 0.203.148 with Kotlin 1.4

**Comment by abrooksv** [comment]
@mfilippov We still have to compile for Rider 193, 201, and 202 so 1.4 may be out of scope for a while
